internal/pkg/csumio/csumwriter.go

```diff
@@ -0,0 +1,89 @@
+package csumio
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+const ChecksumLen = 8
+
+type ChecksumWriter struct {
+	dataFile     io.Writer
+	checksumFile io.Writer
+	chunkSize    int
+	buf          []byte
+}
+
+func NewChecksumWriter(dataFile, checksumFile io.Writer, chunkSize int) (*ChecksumWriter, error) {
+	if chunkSize <= 4*1024 && chunkSize%(4*1024) != 0 {
```

Comment:
Please define a constant storing like `minimumChunkSize`.  We can use this value not only here but also in tests.